### PR TITLE
SONARGRADL-6 Add automatic support for Android projects

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarqube/gradle/AndroidUtils.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    properties.put(SonarQubePlugin.SONAR_JAVA_TARGET_PROP, releaseVariant.getJavaCompiler().getTargetCompatibility());
+
+    List<File> libraries = new ArrayList<>();
+    libraries.addAll(bootClassPath);
+    libraries.addAll(releaseVariant.getJavaCompiler().getClasspath().getFiles());
+    SonarQubePlugin.setMainClasspathProps(properties, false, releaseVariant.getJavaCompiler().getDestinationDir(), libraries);
+    return bootClassPath;
+  }
+
+  private static List<File> getFilesFromSourceSet(SourceProvider sourceSet) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/AndroidUtils.java]: MAJOR Remove this unused private "getFilesFromSourceSet" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/main/java/org/sonarqube/gradle/AndroidUtils.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.stream.Collectors;
+import org.gradle.api.Project;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Only access this class when running on an Android application
+ */
+class AndroidUtils {
+
+    static final String RELEASE = "release";
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/AndroidUtils.java]: MAJOR Make this line start at column 3. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 3]
```diff
@@ -168,9 +179,17 @@ private void addGradleDefaults(final Project project, final Map<String, Object>
...
     configureForGroovy(project, properties);
 
+    if (isAndroidProject(project)) {
+      AndroidUtils.configureForAndroid(project, properties);
+    }
+
     properties.putIfAbsent(SONAR_SOURCES_PROP, "");
   }
 
+  private boolean isAndroidProject(Project project) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: MAJOR Make "isAndroidProject" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 4]
```diff
@@ -168,9 +179,17 @@ private void addGradleDefaults(final Project project, final Map<String, Object>
...
 
+    if (isAndroidProject(project)) {
+      AndroidUtils.configureForAndroid(project, properties);
+    }
+
     properties.putIfAbsent(SONAR_SOURCES_PROP, "");
   }
 
+  private static boolean isAndroidProject(Project project) {
+    return project.getPlugins().hasPlugin("com.android.application") || project.getPlugins().hasPlugin("com.android.library")|| project.getPlugins().hasPlugin("com.android.test");
```
Comment (dbmeneses) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: do they all provide com.android.tools.build:gradle?

**Comment by dbmeneses** [comment]
LGTM

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java (hunk 4 above)
Comment (henryju) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: I hope so ;) This was suggested by Gradle guys: https://discuss.gradle.org/t/improve-sonarqube-plugin-to-support-android-projects/12368/2?u=henryju

**Pull request merged**